reinvent/runmodes/RL/run_staged_learning.py

```diff
@@ -45,7 +45,7 @@ def setup_diversity_filter(config: SectionDiversityFilter, rdkit_smiles_flags: d
 
     memory_type = config.type
 
-    if "type" in config:
+    if memory_type:
```

Comment:
A robust test would need to check if `config `actually has a `type `attribute i.e. test for `hasattr(config, "type")`

Reply:
This is indeed better and it is working, I made the change to "hasattr(config,"type") in my pull request